packages/x-date-pickers/src/internals/hooks/usePickerState.ts

```diff
@@ -69,99 +151,98 @@ export const usePickerState = <TInput, TDateValue>(
     }
   }, [parsedDateValue]);
 
-  const [draftState, dispatch] = React.useReducer(
-    (state: Draftable<TDateValue>, action: DraftAction<TDateValue>): Draftable<TDateValue> => {
-      switch (action.type) {
-        case 'reset':
-          return initDraftableDate(action.payload);
-        case 'update':
-          return {
-            ...state,
-            draft: action.payload,
-          };
-        default:
-          return state;
-      }
-    },
-    parsedDateValue,
-    initDraftableDate,
-  );
-  if (!valueManager.areValuesEqual(utils, draftState.committed, parsedDateValue)) {
-    dispatch({ type: 'reset', payload: parsedDateValue });
-  }
-
-  const [initialDate, setInitialDate] = React.useState<TDateValue>(draftState.committed);
-
-  // Mobile keyboard view is a special case.
-  // When it's open picker should work like closed, because we are just showing text field
-  const [isMobileKeyboardViewOpen, setMobileKeyboardViewOpen] = React.useState(false);
+  React.useEffect(() => {
+    if (isOpen) {
+      // Update all dates in state to equal the current prop value
+      setDate({ action: 'setAll', value: parsedDateValue, internal: true });
+    }
+  }, [isOpen]); // eslint-disable-line react-hooks/exhaustive-deps
 
-  const acceptDate = React.useCallback(
-    (acceptedDate: TDateValue, needClosePicker: boolean) => {
-      onChange(acceptedDate);
+  // Set the draft and committed date to equal the new prop value.
+  if (!valueManager.areValuesEqual(utils, dateState.committed, parsedDateValue)) {
+    setDate({ action: 'setCommitted', value: parsedDateValue, internal: true });
+  }
 
-      if (needClosePicker) {
-        setIsOpen(false);
-        setInitialDate(acceptedDate);
-        if (onAccept) {
-          onAccept(acceptedDate);
-        }
-      }
-    },
-    [onAccept, onChange, setIsOpen],
-  );
+  const shouldCloseOnSelect = !(disableCloseOnSelect ?? wrapperVariant === 'mobile');
```

Comment:
Now that you put a default value with `disableCloseOnSelect = wrapperVariant === 'mobile'` I assume you can remove the `??`

Reply:
Users should still be able to override it right ?